proposal.rst

```diff
@@ -271,7 +309,34 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+Community Bonding Period (May 4 - May 31)
+    * Interact with mentors, introduce myself to the community.
+    * Get familiar with Xapian’s development style. Understand expected coding, documentation, and testing standards set by Xapian.
+    * Start working on some issues to familiarise with the codebase. (Not necessarily related to the project)
+    * Discuss and prepare a list of various MIME types that need to be worked on.
+    * Read documentation relating to the required libraries.
+
+First Coding Period (June 1 - June 28)
+    * Implement first handler, update build system, and add new worker to omindex. (7 days)
+    * Look for issues, take feedback, improve performance. (6 days)
+    * Document the code. (4 days)
+    * Finalise the PR and merge. (2 days)
```

Comment:
From experience, your first PR of GSoC will almost certainly take longer than that, particularly if you haven't had a PR merged before or during the community bonding period. You can (and should) overlap dealing with rounds of feedback on your first PR with later work, but the priority should always be getting the PR merged (otherwise you can end up with multiple open PRs, none ready to merge).

Reply:
Thanks for the feedback. The time taken in feedback and resolving issues is the effective time. It will happen concurrently with the next task.